src/commonTest/kotlin/fr/acinq/lightning/wire/InitTlvTestsCommon.kt

```diff
@@ -0,0 +1,33 @@
+package fr.acinq.lightning.wire
+
+import fr.acinq.bitcoin.PublicKey
+import fr.acinq.lightning.tests.utils.LightningTestSuite
+import fr.acinq.secp256k1.Hex
+import kotlin.test.Test
+import kotlin.test.assertEquals
+
+class InitTlvTestsCommon : LightningTestSuite() {
+    @Test
+    fun `legacy phoenix TLV`() {
+        val testCases = listOf(
+            Pair(PublicKey(Hex.decode("03864ef025fde8fb587d989186ce6a4a186895ee44a926bfc370e2c366597a3f8f")), Hex.decode("fe47020001 21 03864ef025fde8fb587d989186ce6a4a186895ee44a926bfc370e2c366597a3f8f"))
+        )
+
+        @Suppress("UNCHECKED_CAST")
+        val readers = mapOf(
+            InitTlv.PhoenixAndroidLegacyNodeId.tag to InitTlv.PhoenixAndroidLegacyNodeId.Companion as TlvValueReader<InitTlv>
+        )
+        val tlvStreamSerializer = TlvStreamSerializer(false, readers)
+
+        testCases.forEach {
+            val decoded = tlvStreamSerializer.read(it.second)
+            val legacyNodeId = decoded.records.mapNotNull { record ->
+                when (record) {
+                    is InitTlv.PhoenixAndroidLegacyNodeId -> record.legacyNodeId
```

Comment:
adding a check on the record's tag here (for example `assertEquals(InitTlv.PhoenixAndroidLegacyNodeId.tag, record.tag`) would fail (see @t-bast's comment above)

Reply:
Oh right, the test passes because when we register the `reader` we correctly use the object tag, it's only on the instances that the tag is incorrect

Reply:
This could be detected automatically if we modify `TlvStreamSerializer.readTlvs()` and check that the tag of the record that was deserialized matches the tag that was read from the input stream and used to select a reader in the internal reader map ?

Reply:
Done with 6f2bee6, I'll let the modification to `TlvStreamSerializer.readTlvs()` for another PR.